osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -1207,6 +1184,43 @@ public override void AddItem(CarouselItem i)
                 base.AddItem(i);
             }
 
+            /// <summary>
+            /// A special method to handle replace operations (general for updating a beatmap).
+            /// Avoids event-driven selection flip-flopping during the remove/add process.
+            /// </summary>
+            /// <param name="oldItem">The beatmap set to be replaced.</param>
+            /// <param name="newItems">All new items to replace the removed beatmap set.</param>
+            /// <returns>All removed items, for any further processing.</returns>
+            public IEnumerable<CarouselBeatmapSet> ReplaceItem(BeatmapSetInfo oldItem, List<CarouselBeatmapSet> newItems)
+            {
+                var previousSelection = (LastSelected as CarouselBeatmapSet)?.Beatmaps
```

Comment:
this is generally unrelated commentary but the fact that this class has both `LastSelected` and `selectedBeatmapSet` that are separately managed and do not delegate to one another is... just great

Reply:
Do you mean the parent class in this case?

Reply:
yeah. i'm not saying anything has to be done about it but yeah